### PR TITLE
Restrict mapping to the handler method to target and triggered element in @HxRequest

### LINKED ISSUE #79: Support for trigger name in `@HxRequest`
When you have a UI that consists of several fragments, that could each trigger an HTMX request, it would be nice if there was a way to disambiguate which element triggered the request without having to resort to additional URIs:

```html
<div>
  <div hx-get="/" name="foo" th:fragment="foo"></div>
  <div hx-get="/" name="bar" th:fragment="bar"></div>
</div>
```

Let's say I have a controller and would like to handle these requests in separate controller methods, the only way would be to change the template to issue the request to different URIs that pollute the URI space. Alternatively, one would have to manually declare a header condition to the request mapping

```java
@HxRequest
@GetRequest(path = "/", headers="HX-Trigger-Name=foo")
…
```

However, HTMX sends `HX-Trigger-Name` headers containing the element triggering the request with it. It would be nice if I could use that name in `@HxRequest` to constrain the mapping like this:

```java
@Controller
class MyController {

  @HxRequest("foo") // Alternatively @HxRequest(trigger = "foo") or @HxRequest(triggerName = "foo")
  String foo() { … }

  @HxRequest("bar")
  String bar() { … }
}
```

As the trigger name is pretty core to identifying an HTMX request, I think it's reasonable to default the (still untaken) `value` attribute to it.

### PR DESCRIPTION
**Pull request by xhaggi**
Closes #79.

### CODE REVIEW

**Comment by checketts** [comment]
Is any work need to ensure this annotation can be composed/subclassed? For example if I wanted to create a `@HxTriggeredBy("someId")` does this approach allow for that?

**Comment by xhaggi** [comment]
@checketts thanks for pointing that out. I changed the code from `AnnotationUtils.findAnnotation` to `AnnotatedElementUtils.findMergedAnnotation` which supports [@AliasFor](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/annotation/AliasFor.html) semantics and annotation hierarchy.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRequest.java [hunk 1]
```diff
@@ -6,9 +6,41 @@
...
+     * Restricts the mapping to the {@code name} of a specific triggered element.
+     *
+     * @see <a href="https://htmx.org/reference/#request_headers">HX-Trigger-Name</a>
+     */
+    String triggerName() default "";
+
+    /**
+     * Restricts the mapping to the {@code id}, if any, or to the {@code name} of a specific triggered element.
+     * <p>
+     * If only one of them should be restricted, use the {@link #triggerId()} or {@link #triggerName()} parameters instead.
```
Comment (wimdeblauwe) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRequest.java:39]: Is this actually the case? I believe the `value` proposal was there as a convenience, but it should be ok I think to just use `@HxRequest("someId")` and `someId` is never used as a name. I see `value` just as an alias to either `triggerId` or `triggerName`. If you want to be explicit use `triggerId` or `triggerName`, but if it is clear enough for you and your project, you can use `value`. Or do you see it differently?

Comment (xhaggi) [reply htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRequest.java:39]: No. I'll adjust the javadoc.

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRequest.java [hunk 2]
```diff
@@ -6,9 +6,41 @@
 import java.lang.annotation.Target;
 
 /**
- * Annotation for mapping htmx requests onto specific handler methods.
+ * Annotation for mapping htmx requests onto specific handler method.
  */
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Retention(RetentionPolicy.RUNTIME)
 public @interface HxRequest {
```
[the remark this answers is not in the document]
Comment (xhaggi) [reply htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HxRequest.java:13]: Sure.

**Comment by xhaggi** [comment]
@wimdeblauwe as already expressed as an idea, I introduced `@HtmxRequestMapping` besides `@HxRequest` and marked `@HxRequest` as deprecated. Let me know if we should go that way. I will adjust the README afterwards.

**Comment by wimdeblauwe** [comment]
I am not a fan of `@HtmxRequestMapping`, because then the code will look like this:

```
@HtmxRequestMapping("my-button")
@GetMapping("/some-url")
    public String doSomething() {
...
}
```

In the case of `@GetMapping` (which is a `@RequestMapping` under the hood), the value indicates the URL path the method is available on. In the case of `@HtmxRequestMapping`, it is not about the URL, but a `triggerId` or `triggerName`. @odrotbohm how do you feel about this?

**Comment by xhaggi** [comment]
IMO it does the same as `@RequestMapping` but for htmx requests, it maps an htmx request to a handler method, optionally with further restrictions. But we can stick with the current naming. It was just an idea.

**Comment by odrotbohm** [comment]
I must have lost track on this conversation (PTO currently), but I agree with Wim, that we shouldn't make this a `…Mapping` annotation. I'm fine with a switch from `@Hx…` to `@Htmx…`. Conceptually, what `@HxRequest` currently does is, is that it *qualifies* the sibling `@…Mapping` annotation, replacing a raw `headers="…"` with a first-class representation of the HTMX concepts that further *constrain* the mapping. The annotation abstracts the implementation (using HTTP headers) that HTMX uses, and I like that. That said, I don't think we should imply that we introduce a parallel way of mapping, as that's not what we do here.

**Comment by xhaggi** [comment]
Ok, thanks, then I'll discard it and we'll keep HxRequest.

**Comment by xhaggi** [comment]
@wimdeblauwe skipped the `HtmxRequestMapping` idea and addressed all your comments.

**Pull request merged**